hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -148,12 +148,11 @@ public void run() {
           }
         }
       } catch (IOException e) { // stream related
+        if (!handleEofException(e)) {
+          LOG.error("Failed to read stream of replication entries", e);
+        }
```

Comment:
The patch looks good to me too.
I had one non patch related question. Does it even make sense to sleep if we were able to handle eof exception successfully ?

Reply:
Nice question! We should skip sleep if the exception can be handled. I modified it.
Thanks for reviwing.